Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM rust:latest
+
+WORKDIR /app/bifrost
+
+COPY . .
```

Comment:
`cargo build` generates a lot of temporary build products, that we don't want in the docker output. 

I've found a (slightly complicated) example here:

  https://docs.docker.com/language/rust/develop/#run-a-database-in-a-container

I think we could simplify it to something like this:

```
################################################################################
# Create a stage for building the application.

ARG RUST_VERSION=1.80.1
FROM rust:${RUST_VERSION}-slim-bullseye AS build
WORKDIR /app

RUN --mount=type=bind,source=src,target=src \
    --mount=type=bind,source=Cargo.toml,target=Cargo.toml \
    --mount=type=bind,source=Cargo.lock,target=Cargo.lock \
    <<EOF
set -e
cargo build --locked --release
cp target/release/bifrost /bifrost
EOF


################################################################################
# Build the final docker image

FROM debian:bullseye-slim AS final

# Copy the executable from the "build" stage.
COPY --from=build /bifrost /app/bifrost

# Expose the port that the application listens on.
EXPOSE 80
EXPOSE 443

WORKDIR /app

# What the container should run when it is started.
CMD ["/app/bifrost"]
```

This also solves another important issue, which is that we need to build in release mode (using `--release`). Otherwise, the binary will be much bigger and slower :)